hasher-matcher-actioner/hmalib/lambdas/api/middleware.py

```diff
@@ -63,9 +64,24 @@ def from_dict(cls, d: t.Dict) -> "DictParseable":
         raise NotImplementedError
 
 
+class DictParseableWithSignalTypeMapping:
+    """
+    A special case of dict parseables where the dynamic signal type config is
+    required.
+    """
+
+    @classmethod
+    def from_dict(
+        cls, d: t.Dict, signal_type_mapping: HMASignalTypeMapping
+    ) -> "DictParseableWithSignalTypeMapping":
+        raise NotImplementedError
+
+
 def jsoninator(
```

Comment:
Lol wut is this name?

Reply:
Oh, first time you are coming across this. :P 

JSON + terminator. I'm really horrible with portmanteaus.

Reply:
Try speaking this name out loud. It gets funnier.